src/snowflake/snowpark/modin/plugin/_internal/align_utils.py

```diff
@@ -92,4 +100,315 @@ def align_axis_0_right(
+    frame2: InternalFrame,
+    join: Literal["inner", "outer", "left", "right"],
+) -> tuple[InternalFrame, InternalFrame]:
+    """
+    Aligns frames on their columns.
+
+    Args:
+        frame1: First frame
+        frame2: Second frame
+        join: How to handle column index
+            'inner': Output frame contains only overlapping columns from both frames.
+            'outer': Output frame contains union of columns from both frames.
+            'left': Output frame contains columns from left frame.
+            'right': Output frame contains columns from right frame.
+
+    Returns:
+        tuple representing aligned left and right InternalFrames.
+    """
+    columns1 = frame1.data_columns_index
+    columns2 = frame2.data_columns_index
+
+    inner_data_column_labels = columns1.intersection(columns2, sort=False).tolist()
+
+    frame1_data_column_pandas_labels = frame1.data_column_pandas_labels
+    frame2_data_column_pandas_labels = frame2.data_column_pandas_labels
+    full_data_column_pandas_labels = get_full_label_list(
+        frame1_data_column_pandas_labels,
+        frame2_data_column_pandas_labels,
+        inner_data_column_labels,
+        join=join,
+    )
+
+    if join == "right":
+        frame1 = align_axis_1_right_helper(
+            frame1, full_data_column_pandas_labels, frame1_data_column_pandas_labels
+        )
+        frame2 = align_axis_1_left_helper(
+            frame2, full_data_column_pandas_labels, frame1_data_column_pandas_labels
+        )
+    else:
+        frame1 = align_axis_1_left_helper(
+            frame1, full_data_column_pandas_labels, frame2_data_column_pandas_labels
+        )
+        frame2 = align_axis_1_right_helper(
+            frame2, full_data_column_pandas_labels, frame2_data_column_pandas_labels
+        )
+    return frame1, frame2
+
+
+def align_axis_1_left_helper(
+    frame: InternalFrame,
+    data_column_labels: list[str],
+    other_frame_labels: list[str],
+) -> InternalFrame:
+    """
+    Select the given labels from data_column_labels for aligned left frame. If any data column label is missing
+    in frame add new column with NULL values. Duplicate column names will also be duplicated.
+
+    Args:
+        frame: An InternalFrame
+        data_column_labels: A list of pandas labels.
+        other_frame_labels: list of other frame labels
+
+    Returns:
+        New InternalFrame representing left aligned frame.
+
+    """
+    select_list: list[ColumnOrName] = []
+
+    # Add index and ordering columns
+    select_list.extend(frame.index_column_snowflake_quoted_identifiers)
+    select_list.extend(frame.ordering_column_snowflake_quoted_identifiers)
+
+    data_column_snowflake_identifiers = []
+    other_counter = Counter(other_frame_labels)
+
+    snowflake_ids = frame.get_snowflake_quoted_identifiers_group_by_pandas_labels(
+        data_column_labels, include_index=False
+    )
+
+    curr_label_count_map: dict[Hashable, int] = {}
```

Comment:
can you use a `Counter` for these variables as well?

Reply:
Sure can do, just curious is this due to better performance over dict?

Reply:
I don't think so (it seems to just use a dict under the hood), I was just asking for consistency with the rest of the PR and so you can remove some `curr_label_count_map.get(label, 0)` boilerplate.

Reply:
true, it should be refactored to make use of counter now